generators/base/generator-base.mjs

```diff
@@ -2446,7 +2448,10 @@ templates: ${JSON.stringify(existingTemplates, null, 2)}`;
     dest.buildToolGradle = dest.buildTool === GRADLE;
     dest.buildToolMaven = dest.buildTool === MAVEN;
     dest.buildToolUnknown = !dest.buildToolGradle && !dest.buildToolMaven;
-    dest.buildDir = this.getBuildDirectoryForBuildTool(dest.buildTool);
+
+    dest.temporaryDir = dest.buildToolGradle ? 'build/' : 'target/';
+    const buildDestinationDir = `${dest.temporaryDir}${dest.buildToolGradle ? 'resources/main/' : 'classes/'}`;
```

Comment:
same remark regarding the logic

Reply:
The entire loadDerivedServerConfig should be moved to generators/server/support in another PR.